packages/backend/src/controllers/fanart.ts

```diff
@@ -31,20 +30,20 @@ export const getAllFanartController = async (req: Request, res: Response) => {
 export const createFanartController = async (req: Request, res: Response) => {
   try {
     // Request body validation
-    const { creator, imageUrl, captchaToken } = req.body
-
-    if (!(await verifyCaptchaToken(captchaToken))) {
-      throw new TypeError('Invalid captcha token')
-    }
-
+    const { creator, imageUrl } = req.body
     if (!(creator && imageUrl))
       throw new TypeError('creator and imageUrl is required')
     if (!isString(creator)) throw new TypeError('creator must be a string')
     if (!isString(imageUrl)) throw new TypeError('imageUrl must be a string')
 
-    const fanart: FanartDoc = await storeFanart({ creator, imageUrl })
+    // Discard records bigger than 4KB (longest possible URL + 2KB for creator)
+    if (creator.length + imageUrl.length > 4096) {
+      throw new TypeError('request body is too large')
```

Comment:
maybe change the error message to a more descriptive message?
e.g `your URL is too long`

Reply:
These messages aren't user facing

Reply:
If I'm not mistaken, currently all error in controller is sent to client side, and in form, these error messages is shown to user

Reply:
Because of how errors are emitted from requests, only 5xx errors and the 429 I explicitly catch should result in the client seeing them. I'll check to be sure though.

Reply:
Confirmed, is not user facing. Browser console logs the resulting 400 but it otherwise fails silently, which was what I intended, as I don't think even the overzealous user should ever hit these limits.